src/persist/src/client.rs

```diff
@@ -423,31 +444,47 @@ impl MultiWriteHandle {
     {
         // Manually pop the first handle to construct the MultiWriteHandle, then
         // continue with the same iter to add the rest.
-        let first_handle = handles
-            .next()
-            .ok_or_else(|| Error::from("MultiWriteHandle requires at least one stream"))?;
-        let mut ret = Self::new(first_handle.borrow())?;
+        let first_handle = match handles.next() {
+            Some(handle) => handle,
+            None => {
+                // TODO: I could change the interface to take always take one stream plus an Iterator.
+                panic!("MultiWriteHandle requires at least one stream");
```

Comment:
does it clean any of these semantics up if instead of separate `client` and `error` fields, we instead have a single `client: Result<RuntimeClient, Error>` one?

Reply:
It does! And I did that before on other handles. I don't know why I didn't think to do it here ... :see_no_evil: